_docs/technical/protocols/2017-01-04-p2p.md

```diff
@@ -219,10 +219,32 @@ a hash, assigning yourself an arbitrary ID is impossible, and this means that a
 
 #### Implementation Notes
 
-[//]: # TODO (CSLD-6): Explain used crypto algorithms in details
+`HashNodeId` is a binary string with a fixed length (32 bytes) formed like this:
+
+~~~
++---------------+------------+
+|    Hashing    |    Nonce   |
++---------------+------------+
+
+|   18 bytes    |  14 bytes  |
+~~~
+
+where:
+
+* `Nonce` is just a random 14 bytes (from the system source of entropy),
+* `Hashing` is a hashing data.
+
+This hashing data generates based on `DerivingKey` and `Salt`, where:
```

Comment:
Does the hashing data generate something, or is it generated? This sentence makes it unclear.